ch_tools/monrun_checks/ch_orphaned_objects.py

```diff
@@ -0,0 +1,159 @@
+from datetime import timedelta
+from tempfile import TemporaryFile
+from typing import Optional
+
+import click
+
+from ch_tools.chadmin.internal.object_storage.obj_list_item import ObjListItem
+from ch_tools.chadmin.internal.object_storage.s3_cleanup import (
+    cleanup_s3_object_storage,
+)
+from ch_tools.chadmin.internal.object_storage.utils import (
+    DEFAULT_GUARD_INTERVAL,
+    get_orphaned_objects_query,
+    get_remote_data_paths_table,
+    get_traverse_shadow_settings,
+    traverse_object_storage,
+)
+from ch_tools.chadmin.internal.utils import execute_query
+from ch_tools.common.cli.parameters import TimeSpanParamType
+from ch_tools.common.clickhouse.config import get_clickhouse_config
+from ch_tools.common.clickhouse.config.storage_configuration import S3DiskConfiguration
+from ch_tools.common.result import CRIT, OK, WARNING, Result
+
+DRY_RUN = True
+ON_CLUSTER = True
+CLUSTER_NAME = "{cluster}"
+DIST_NAME = "object_storage"
+OBJECT_NAME_PREFIX = None
+
+
+@click.command("orphaned-objects")
+@click.option(
+    "--keep-paths",
+    "keep_paths",
+    is_flag=True,
+    help=("Do not delete collected paths of objects from object storage."),
+)
+@click.option(
+    "--use-saved-list",
+    "use_saved_list",
+    is_flag=True,
+    help=("Use saved object list without traversing object storage again."),
+)
+@click.option(
+    "-c", "--critical", "crit", type=int, default=3600, help="Critical threshold."
+)
+@click.option(
+    "-w", "--warning", "warn", type=int, default=600, help="Warning threshold."
+)
+@click.option(
+    "--from-time",
+    "from_time",
+    default=None,
+    type=TimeSpanParamType(),
+    help=(
+        "Begin of inspecting interval in human-friendly format. "
+        "Objects with a modification time falling interval [now - from_time, now - to_time] are considered."
+    ),
+)
+@click.option(
+    "-g",
+    "--guard-interval",
+    "--to-time",
+    "to_time",
+    default=DEFAULT_GUARD_INTERVAL,
+    type=TimeSpanParamType(),
+    help=("End of inspecting interval in human-friendly format."),
+)
+@click.pass_context
+def orphaned_objects_command(
+    ctx: click.Context,
+    keep_paths: bool,
+    use_saved_list: bool,
+    crit: int,
+    warn: int,
+    from_time: Optional[timedelta],
+    to_time: timedelta,
+) -> Result:
+    ch_config = get_clickhouse_config(ctx)
```

Comment:
The code of this function is very similar to the code of  `clean_command` from [object_storage_group.py](https://github.com/yandex/ch-tools/pull/185/files#diff-0ce319818fcec2795390bb14e856b3f9d87d4b572bb067b9caf86b4e8dfc760e). And it can be actively changed in the future. It would be great to extract it also to one place.

Reply:
thanks, done